main.go

```diff
@@ -285,3 +305,17 @@ func runAndReportAudit(c conf.Configuration, auditPath string, outputFile string
 	}
 	return auditData
 }
+func decode(oldFileBytes []byte) (validator.AuditData, error) {
```

Comment:
To keep things clean, can you move this function into `pkg/validator/types.go`, and rename it to `ParseAudit`?

Even better, maybe create a `GetAuditFromFile` function in `pkg/validator/types.go` that contains all the logic above in `main()`.

In general, it's good to keep `main.go` small, since it's not as easy to test/reuse as something in `pkg/`. We've let it get a bit out of hand 😬

Reply:
Agreed, the main.go is getting a bit long and has lots of logic in it. I also agree the name the function could use some love. `ParseAudit` is ok with me, but I'm also very verbose when it comes to function names and `ReadAuditFromFile()` usually is something I would be explicit about.